### PR TITLE
You cannot make infinite cloth anymore

### PR DESCRIPTION
**Pull request by Penwin0**
## About The Pull Request
No more infinite cloth

## Why It's Good For The Game
It's an exploit

## Changelog
:cl:
fix: You can't make infinite cloth anymore
/:cl:

### CODE REVIEW

**Comment by Addust** [comment]
tell me how so i can abuse it in the thunderdome to give out cloth until this is merged

**Comment by HourlyLimit** [comment]
indeed we want to know

**Comment by Penwin0** [comment]
You can spam click the item a million times, just hold the pop ups, then click them all afterwards since it currently doesn't check if the item still exists

It's currently testmerged so I am afraid no infinite cloth for you

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -92,6 +92,9 @@
 	if(W.get_sharpness() && cuttable)
 		if (alert(user, "Are you sure you want to cut the [src] into strips?", "Cut clothing:", "Yes", "No") != "Yes")
 			return
+		if(!src)
```
Comment (SuperNovaa41) [inline code/modules/clothing/clothing.dm]: um, no

just `qdel(src)` after cutting it into strips

**Review by SuperNovaa41 — CHANGES_REQUESTED** [review]

File: code/modules/clothing/clothing.dm (hunk 1 above)
Comment (Penwin0) [reply code/modules/clothing/clothing.dm]: That is infact already done at the bottom of this proc

Comment (SuperNovaa41) [reply code/modules/clothing/clothing.dm]: ok then add a `if (QDELETED(src))` check at the very beginning not midway through

Comment (Penwin0) [reply code/modules/clothing/clothing.dm]: It has to be after the alert else it won't work

Comment (Penwin0) [inline code/modules/clothing/clothing.dm]:
```suggestion
		if(QDELETED(src))
```

**Review by SuperNovaa41 — APPROVED** [review]

**Pull request merged**